truss/templates/server/model_wrapper.py

```diff
@@ -307,32 +318,58 @@ def ready(self) -> bool:
     def _model_file_name(self) -> str:
         return self._config["model_class_filename"]
 
-    def start_load_thread(self):
-        # Don't retry failed loads.
-        if self._status == ModelWrapper.Status.NOT_READY:
-            thread = Thread(target=self.load)
-            thread.start()
+    async def setup_environment(self, environment: dict):
+        descriptor = self.model_descriptor.setup_environment
+        if not descriptor:
+            return
+        if descriptor.is_async:
+            return await self._model.setup_environment(environment)
+        else:
+            return await to_thread.run_sync(self._model.setup_environment, environment)
+
+    async def poll_for_environment_updates(self) -> None:
+        last_modified_time = None
+        ENVIRONMENT_CONFIG_FILENAME = "/tmp/environment"
+        SLEEP_TIME_SECONDS = 10
+        while True:
+            if os.path.isfile(ENVIRONMENT_CONFIG_FILENAME):
+                try:
+                    current_mtime = os.path.getmtime(ENVIRONMENT_CONFIG_FILENAME)
+                    if not last_modified_time or last_modified_time != current_mtime:
+                        # read in the file
+                        async with aiofiles.open(ENVIRONMENT_CONFIG_FILENAME, "r") as f:
```

Comment:
should we try to utilize `dynamic_config_resolver` here? we should have access to the `shared` resolvers here

Reply:
I think so right?